tmail-backend/mailbox/team-mailboxes/src/main/scala/com/linagora/tmail/team/TeamMailboxUserEntityValidator.scala

```diff
@@ -0,0 +1,36 @@
+package com.linagora.tmail.team
+
+import com.linagora.tmail.team.TeamMailboxUserEntityValidator.{TEAM_MAILBOX, TEAM_MAILBOX_NAMESPACE}
+import org.apache.james.core.Username
+import org.apache.james.mailbox.{MailboxManager, MailboxSession}
+import org.apache.james.mailbox.model.{MailboxConstants, MailboxPath}
+import org.apache.james.mailbox.store.MailboxReactorUtils.block
+import org.apache.james.webadmin.service.UserEntityValidator
+import org.apache.james.webadmin.service.UserEntityValidator.{EntityType, ValidationFailure}
+
+import java.util
+import javax.inject.Inject
+
+object TeamMailboxUserEntityValidator {
+  val TEAM_MAILBOX: EntityType = new EntityType("team-mailbox")
+
+  val TEAM_MAILBOX_NAMESPACE: String = MailboxConstants.NAMESPACE_PREFIX_CHAR + "TeamMailbox"
+}
+
+class TeamMailboxUserEntityValidator @Inject()(mailboxManager: MailboxManager) extends UserEntityValidator {
+  override def canCreate(username: Username, ignoredTypes: util.Set[EntityType]): Option[UserEntityValidator.ValidationFailure] = {
+    if (ignoredTypes.contains(TEAM_MAILBOX)) {
+      return None
+    }
+
+    if (username.getDomainPart.isPresent) {
+      val mailboxSession: MailboxSession = mailboxManager.createSystemSession(username)
+      val checkPath: MailboxPath = new MailboxPath(TEAM_MAILBOX_NAMESPACE, Username.of(username.getDomainPart.get.asString()), username.getLocalPart + "." + MailboxConstants.INBOX)
```

Comment:
`Username.of(username.getDomainPart.get.asString())`

What if the username do not have a domain part?

Reply:
the purpose of `if (username.getDomainPart.isPresent) {` two lines above?

Reply:
IMO in scala we should make the effort of writing fully functional code.

The main pain toward it in Java is unchecked exceptions. There's not this in Scala.

Reply:
Agreed I know, but I realized I missed the all team mailbox api and that it was useless to put more efforts into this for now... But point taken